### PR TITLE
[imgui] Update to 1.89.9

### PR DESCRIPTION
**Pull request by RT2Code**
Updates imgui port from 1.89.7 to 1.89.9 :
https://github.com/ocornut/imgui/releases/tag/v1.89.9
https://github.com/ocornut/imgui/releases/tag/v1.89.8

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Pospelove** [comment]
from 1.89.8 changelog:

> Fonts, imgui_freetype: Added support to render OpenType SVG fonts using lunasvg. Requires enabling IMGUI_ENABLE_FREETYPE_LUNASVG along with IMGUI_ENABLE_FREETYPE, and providing headers/libraries for lunasvg. (https://github.com/ocornut/imgui/pull/6591, https://github.com/ocornut/imgui/issues/6607) [sakiodre]

could you please add an option for that? vcpkg has lunasvg port

**Comment by RT2Code** [comment]
@Pospelove Done. 😉

**Comment by Pospelove** [comment]
> @Pospelove Done. 😉

Amazing, thank you

**Comment by JonLiu1993** [comment]
@RT2Code, when I tested the usage by `./vcpkg install imgui[freetype-lunasvg]:x64-windows` I get this error:
CMakeLists:
````
cmake_minimum_required (VERSION 3.8)

project(test)

find_package(imgui CONFIG REQUIRED)

# Add source to this project's executable.
add_executable (test "usageTest.cpp")

target_link_libraries(test PRIVATE imgui::imgui)
````
usageTest.cpp:
````
#include <stdio.h>
#include <imgui.h>

int main()
{
    printf("%s\n", "HELLO WORLD");
    return 0;
}
````
config:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/imgui/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32538.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32538.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Found ZLIB: optimized;F:/Feature-test/imgui/vcpkg/installed/x64-windows/lib/zlib.lib;debug;F:/Feature-test/imgui/vcpkg/installed/x64-windows/debug/lib/zlibd.lib (found version "1.3.0")  
1> [CMake] -- Configuring done (13.2s)
1> [CMake] -- Generating done (0.1s)
1> [CMake] -- Build files have been written to: C:/Users/test/source/repos/usageTest/out/build/x64-debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

````
build:
````
>------ Build All started: Project: usageTest, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/test/source/repos/usageTest/usageTest/CMakeLists.txt
    usageTest.cpp
C:\Users\test\source\repos\usageTest\out\build\x64-Debug\usageTest\LINK : fatal error LNK1104: cannot open file 'unofficial-lunasvg.lib' 

Build All failed.
````

File: ports/imgui/CMakeLists.txt [hunk 1]
```diff
@@ -112,6 +112,12 @@ if(IMGUI_FREETYPE)
     target_compile_definitions(${PROJECT_NAME} PUBLIC IMGUI_ENABLE_FREETYPE)
 endif()
 
+if(IMGUI_FREETYPE_LUNASVG)
+    find_package(unofficial-lunasvg CONFIG REQUIRED)
+    target_link_libraries(${PROJECT_NAME} PUBLIC unofficial-lunasvg)
```
Comment (Pospelove) [inline ports/imgui/CMakeLists.txt:117]: I guess it should look like this
```cmake
target_link_libraries(${PROJECT_NAME} PUBLIC unofficial-lunasvg::unofficial-lunasvg)
```

Comment (JonLiu1993) [reply ports/imgui/CMakeLists.txt:117]: I think so.

**Comment by RT2Code** [comment]
Whoops, I had this error too, and fixed it, but it looks like I've somehow forgotten to commit the fix. It should build fine now.

File: ports/imgui/CMakeLists.txt (hunk 1 above)
Comment (RT2Code) [reply ports/imgui/CMakeLists.txt]: Almost, the right target name is unofficial::lunasvg::lunasvg. I love how those namespaces are not confusing at all. ^^

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
feature `freetype-lunasvg` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

Tested usage successfully by `imgui[freetype-lunasvg]:x64-windows`:
````
imgui provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(imgui CONFIG REQUIRED)
    target_link_libraries(main PRIVATE imgui::imgui)
````

**Comment by BillyONeal** [comment]
Thanks for the update!

**Comment by autoantwort** [comment]
When I try to build the new feature I get 
```
Change Dir: '/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/arm64-osx-dbg'

Run Build Command(s): /Users/leanderSchulten/git_projekte/vcpkg/downloads/tools/ninja/1.10.2-osx/ninja -v -v -j11 install
[1/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/imgui_draw.cpp.o -MF CMakeFiles/imgui.dir/imgui_draw.cpp.o.d -o CMakeFiles/imgui.dir/imgui_draw.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui_draw.cpp
[2/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/imgui_tables.cpp.o -MF CMakeFiles/imgui.dir/imgui_tables.cpp.o.d -o CMakeFiles/imgui.dir/imgui_tables.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui_tables.cpp
[3/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/misc/cpp/imgui_stdlib.cpp.o -MF CMakeFiles/imgui.dir/misc/cpp/imgui_stdlib.cpp.o.d -o CMakeFiles/imgui.dir/misc/cpp/imgui_stdlib.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/cpp/imgui_stdlib.cpp
[4/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o -MF CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o.d -o CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/freetype/imgui_freetype.cpp
FAILED: CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o -MF CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o.d -o CMakeFiles/imgui.dir/misc/freetype/imgui_freetype.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/freetype/imgui_freetype.cpp
/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/freetype/imgui_freetype.cpp:66:32: warning: unknown warning group '-Wsubobject-linkage', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wsubobject-linkage"        // warning: 'xxxx' has a field 'xxxx' whose type uses the anonymous namespace
                               ^
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/freetype/imgui_freetype.cpp:36:
/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui.h:1799:57: error: object expression of non-scalar type 'void' cannot be used in a pseudo-destructor expression
template<typename T> void IM_DELETE(T* p)   { if (p) { p->~T(); ImGui::MemFree(p); } }
                                                       ~^
/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/misc/freetype/imgui_freetype.cpp:843:5: note: in instantiation of function template specialization 'IM_DELETE<void>' requested here
    IM_DELETE(*_state);
    ^
1 warning and 1 error generated.
[5/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/imgui_demo.cpp.o -MF CMakeFiles/imgui.dir/imgui_demo.cpp.o.d -o CMakeFiles/imgui.dir/imgui_demo.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui_demo.cpp
[6/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/imgui_widgets.cpp.o -MF CMakeFiles/imgui.dir/imgui_widgets.cpp.o.d -o CMakeFiles/imgui.dir/imgui_widgets.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui_widgets.cpp
[7/9] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DIMGUI_ENABLE_FREETYPE -DIMGUI_ENABLE_FREETYPE_LUNASVG -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean -isystem /Users/leanderSchulten/git_projekte/vcpkg/vcpkg_installed/arm64-osx/include -fPIC -g -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/imgui.dir/imgui.cpp.o -MF CMakeFiles/imgui.dir/imgui.cpp.o.d -o CMakeFiles/imgui.dir/imgui.cpp.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/imgui/src/v1.89.9-35b8d7fda8.clean/imgui.cpp
ninja: build stopped: subcommand failed.
```

**Pull request merged**